src/nextjs-front/components/chat/GroupNew.tsx

```diff
@@ -84,23 +85,28 @@ const GroupNew: React.FC = () => {
 		e.preventDefault();
 		const errors: Partial<NewGroupData> = {};
 
-		formData.groupName = formData.groupName.trim();
+		if (!pendingChanges) return ;
 
-		if (formData.groupName.length < 3 || formData.groupName.length > 20) {
-			errors["groupName"] = "Group name should be between 3 and 20 characters long";
+		const nameLen = formData.groupName.trim().length;
+
+		if (nameLen < 3 || nameLen > 20) {
+			errors["groupName"] = "Name must be 3 to 20 characters long";
+		} else if (!(/^[a-zA-Z0-9_ ]+$/.test(formData.groupName))) {
```

Comment:
I think it would be better to extract the regexp logic inside their own functions.
For example, think about implementing a `isGroupNameValid` or `isGroupPasswordValid` functions.